src/battle/overlay_12_0224E4FC.c

```diff
@@ -3438,3 +3438,429 @@ int DamageDivide(int num, int denom) {
+                            for (index = 0; index < LEARNED_MOVES_MAX; index++) {
+                                moveNo = ctx->battleMons[battlerIdCheck].moves[index];
+                                if (moveNo) {
+                                    moveStatus = 0;
+                                    ctx->damage = ov12_02251D28(bsys, ctx, moveNo, 0, battlerIdCheck, battlerId, ctx->damage, &moveStatus);
+                                    if (!(moveStatus & MOVE_STATUS_NO_EFFECT) && !ov12_0225865C(ctx, moveNo) && ((moveStatus & MOVE_STATUS_SUPER_EFFECTIVE) || (ctx->unk_334.moveData[moveNo].effect == 38 && ctx->battleMons[battlerId].level <= ctx->battleMons[battlerIdCheck].level))) {
+                                        flag = TRUE;
+                                        break;
+                                    }
+                                }
+                            }
+                            if (flag == TRUE) {
+                                break;
+                            }
+                        }
+                    }
+                    if (flag == TRUE) {
+                        ctx->battlerIdWork = battlerId;
+                        script = 194;
+                    }
+                    break;
+                }
+            }
+            if (i == maxBattlers) {
+                ctx->sendOutState++;
+            }
+            break;
+        case 6: //forewarn
+            for (i = 0; i < maxBattlers; i++) {
+                battlerId = ctx->turnOrder[i];
+                if (!ctx->battleMons[battlerId].forewarnFlag && ctx->battleMons[battlerId].hp && (GetBattlerAbility(ctx, battlerId) == ABILITY_FOREWARN)) {
+                    ctx->battleMons[battlerId].forewarnFlag = TRUE;
+                    int battlerIdCheck;
+                    int index;
+                    u16 moveNo;
+                    u32 power;
+                    int hp;
+                    u32 powerTemp;
+                    
+                    
+                    powerTemp = 0;
+                    hp = 0;
+                    
+                    for (battlerIdCheck = 0; battlerIdCheck < maxBattlers; battlerIdCheck++) {
+                        if (BattleSystem_GetFieldSide(bsys, battlerId) != BattleSystem_GetFieldSide(bsys, battlerIdCheck) && ctx->battleMons[battlerIdCheck].hp) {
+                            hp += ctx->battleMons[battlerIdCheck].hp;
+                            for (index = 0; index < LEARNED_MOVES_MAX; index++) {
+                                moveNo = ctx->battleMons[battlerIdCheck].moves[index];
+                                power = ctx->unk_334.moveData[moveNo].power;
+                                switch (power) {
+                                case 1:
+                                    switch (ctx->unk_334.moveData[moveNo].effect) {
+                                    case 38: //OHKO?
```

Comment:
constants?

Reply:
I'm only sure about the OHKO one, gonna wait till I can get more info about move effects before documenting these